PowerPointLabs/PowerPointLabs/AudioMisc/Audio.cs

```diff
@@ -41,6 +45,27 @@ public Audio(string name, string saveName, int matchScriptID)
             Type = AudioHelper.GetAudioType(saveName);
         }
 
+        public Audio(Shape shape, string saveName)
+        {
+            // detect audio type from shape tag
+            AudioType audioType;
+            if (!Enum.TryParse<AudioType>(shape.Tags[AudioTypeTagName], out audioType))

```

Comment:
the audio shape created previously may not have the tag, and then it will be treated as `unrecognized`. pls make sure it's backward compatible